quark/evaluator/pyeval.py

```diff
@@ -177,6 +183,12 @@ def _invoke(self, instruction, look_up=False, skip_self=False):
                 var_obj = self.table_obj.pop(index)
                 value_of_reg_list.append(var_obj.value)
 
+        # Remove duplicate parameter values to save memory
```

Comment:
Replaced the nested loops with a dictionary-based approach to track seen values.

For example.

```Python
seen = {} 
for idx, val in enumerate(value_of_reg_list): 
    if val in seen: 
        value_of_reg_list[idx] = f"<ref_{seen[val]}>" 
    else: 
        seen[val] = idx
```